### ISSUE TITLE
SteamVR beta is not launching

### ISSUE THREAD

**Issue by dylanmtaylor** [ValveSoftware/SteamVR-for-Linux#618]
#### Your system information

* Steam client version (build number or date):  
Steam Version:  1695334486
Steam Client Build Date:  Thu, Sep 21 11:13 AM UTC -05:00
Steam Web Build Date:  Thu, Sep 21 5:49 PM UTC -05:00
Steam API Version:  SteamClient020
* Distribution (e.g. Ubuntu):  Ubuntu 23.10 beta
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes
* Steam Logs: [generate by running this command in a terminal `tar -zcvf ~/Desktop/steam-logs.tar.gz ~/.steam/steam/logs`]
* GPU: Nvidia RTX 3090

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

When attempting to launch SteamVR (beta), I see this in the logs.
```
vrsetup.sh[4027713]: exec /home/dylan/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh /home/dylan/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
/home/dylan/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 30: cd: /home/dylan/.steam/debian-installation/steamapps/common/SteamVR/../runtime: No such file or directory
/home/dylan/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 31: cd: /home/dylan/.steam/debian-installation/steamapps/common/SteamVR/../sdk: No such file or directory
vrenv.sh[4027713]: exec /home/dylan/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
vrsetup.sh[4027713]: /home/dylan/.steam/debian-installation/steamapps/common/SteamVR/bin/linux64/vrcompositor-launcher binary has cap_sys_nice privileges
```

Nothing happens visually when I click the button to open SteamVR.

#### Steps for reproducing this issue:

I'm not sure how I got it to this state. I can't get VR working on Linux.

**Comment by mSparks43** [comment]
SteamVR/bin/vrstartup.sh to launch steamVR
vrsetup.sh just (afaik) installs stuff as root for VR applications to find it.

**Comment by dylanmtaylor** [comment]
I am launching it through the steam client, and these are the error messages that appear in my console.

**Comment by ThaSwapMeetPimp** [comment]
Am having same issue on Ubuntu-Studio, though all I can find in the logs is that its not even generating any entries, though I could be looking in the wrong ones.

**Comment by JupiterSky11** [comment]
Also having this issue on Ubuntu 22.04.  `vrsetup.sh` just hangs, doesn't respond to signals either.

**Comment by ClearlyClaire** [comment]
The same occurs to me. The script executes once and executes `exec steam-runtime-launch-client --alongside-steam "$0" "$@"`; then the script is executed again in its new environment, running to completion, but `steam-runtime-launch-client` hangs with 100% CPU usage.

**Comment by LiamDawe** [comment]
Also seeing this exact same problem on Kubunt 23.04, it just doesn't launch.

https://gist.github.com/LiamDawe/2ae46d1b4862a142a9e9fb926f8db381

**Comment by JupiterSky11** [comment]
I "fixed" it by modifying the bash file and moving the last couple lines of `vrstartup.sh` to the end of `vrsetup.sh`, however there were other issues with `vrenv.sh` and upon the program starting it "couldn't connect to the window manager" with error code 497.

**Comment by JupiterSky11** [comment]
Still broken.  Anyone have this in their logs?
```
/home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 30: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../runtime: No such file or directory
/home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 31: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../sdk: No such file or directory
```

**Comment by jarettmillard** [comment]
I was not affected by this issue in earlier 2.x betas, but I am now with the latest 2.0.3 beta. Running Pop OS 22.04.

**Comment by LiamDawe** [comment]
Still not working on Kubuntu with the latest update as well: https://gist.github.com/LiamDawe/83fabf58aa729d6d94d9506a530ec91f

**Comment by extraymond** [comment]
> Still broken. Anyone have this in their logs? I see that LiamDawe has the same error.
> 
> ```
> /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 30: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../runtime: No such file or directory
> /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 31: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../sdk: No such file or directory
> ```
> 
> I tried to find where these are supposed to point, but no dice. There aren't any files or folders named "sdk" anywhere in the steam directory, and runtime exists in a few cases but not verbosely.

+1 
Ubuntu 22.04 with steam client beta and steamvr beta, having the same issue.
It launch with stable steam client accompanied with steamvr beta though, albeit only able to see the old ui.

**Comment by digitalcircuit** [comment]
I reported the same issue on the SteamVR "Bug Reports" forum and didn't think to check the SteamVR for Linux issue tracker.  [The details and log files I shared there look the same as what's here](https://steamcommunity.com/app/250820/discussions/3/3887227031498631621/ ), but I figured I should link it here so Valve/etc know to close that report when this one is closed.

**Comment by ThaSwapMeetPimp** [comment]
> > Still broken. Anyone have this in their logs? I see that LiamDawe has the same error.
> > ```
> > /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 30: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../runtime: No such file or directory
> > /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 31: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../sdk: No such file or directory
> > ```
> > 

    Still broken. Anyone have this in their logs? I see that LiamDawe has the same error.

    /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 30: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../runtime: No such file or directory
    /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 31: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../sdk: No such file or directory

    I tried to find where these are supposed to point, but no dice. There aren't any files or folders named "sdk" anywhere in the steam directory, and runtime exists in a few cases but not verbosely.

+1
Ubuntu 22.04 with steam client beta and steamvr beta, having the same issue.
It launches with stable steam client accompanied with steamvr beta though, albeit only able to see the old ui.

> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I tried to find where these are supposed to point, but no dice. There aren't any files or folders named "sdk" anywhere in the steam directory, and runtime exists in a few cases but not verbosely.
> 
> +1 Ubuntu 22.04 with steam client beta and steamvr beta, having the same issue. It launches with stable steam client accompanied with steamvr beta though, albeit only able to see the old ui.

Have you tried creating the folders that are missing?  My system does that, at some point when using commands like mkdir I started having to mkdir each missing location in the location tree (so to make /media/username/location, if username and location didnt exist, I cant just type out 'mkdir /media/username/location' anymore, now I have to type out 'mkdir /media/username' then 'mkdir /media/username/location'. The only reason I havent tried it myself is the /../ in the location tree, is that /../ indicating other folders in the tree or a folder that is labeled ..

**Comment by ThaSwapMeetPimp** [comment]
> Still broken. Anyone have this in their logs? I see that LiamDawe has the same error.
> 
> ```
> /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 30: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../runtime: No such file or directory
> /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 31: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../sdk: No such file or directory
> ```
> 
> I tried to find where these are supposed to point, but no dice. There aren't any files or folders named "sdk" anywhere in the steam directory, and runtime exists in a few cases but not verbosely.

Have you tried creating the folders that are missing?  My system does that, at some point when using commands like mkdir I started having to mkdir each missing location in the location tree (so to make /media/username/location, if username and location didnt exist, I cant just type out 'mkdir /media/username/location' anymore, now I have to type out 'mkdir /media/username' then 'mkdir /media/username/location'). The only reason I havent tried it myself is the /../ in the location tree, is that /../ indicating other folders in the tree or a folder that is labeled ..

**Comment by mSparks43** [comment]
>/home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 30: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../runtime: No such file or directory
/home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 31: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../sdk: No such file or directory

These are missing on a normal working install of 1.27.5

**Comment by ThaSwapMeetPimp** [comment]
> > /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 30: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../runtime: No such file or directory
> > /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh: line 31: cd: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/../sdk: No such file or directory
> 
> These are missing on a normal working install of 1.27.5

Whats your definition of a normal working install? There are things that dont work reliably or at all (at least for me) on Linux that do on Windows (Desktop and Desktop windows in VR Space, for example), maybe the new beta relies on those to work, and those missing directories are part of why they dont work. Maybe its hanging on a part that checks for those.

Im just throwing things at the wall here though, I dont really know

**Comment by mSparks43** [comment]
> Whats your definition of a normal working install? 
#609 

>There are things that dont work reliably or at all (at least for me) on Linux that do on Windows (Desktop and Desktop windows in VR Space, for example), maybe the new beta relies on those to work, and those missing directories are part of why they dont work. Maybe its hanging on a part that checks for those.
> 
> Im just throwing things at the wall here though, I dont really know

Beta 2 doesn't launch for me either, I've moved 1.27.5 to an offline version and sticking with that for now, all I need it for is to connect to the headset, it does that. I'm bored of being terrified every update will nuke my "working" install even if there are still plenty of issues to fix in the UI. I'm trying to develop for VR and have already lost months if not years to forced versions that won't start or CTD. I have zero faith this wont be made "final" in its current state. Last time they removed the compositor window which was a godsend for developing without a headset connected

**Comment by ThaSwapMeetPimp** [comment]
Still not launching with 2.0.4, and trying on Steam Client Stable Channel I get error 497 Failed to connect to Window Manager.

**Comment by ThaSwapMeetPimp** [comment]
Am i the only one who is feeling like they may have just borked it on purpose? Like an F you to people using distros other than what SteamOS is on? Which they still havent released 3.x for PC, even though they said they would. I mean its been 3 weeks since OP posted and its not even marked as a bug yet. See multiple people saying they have the issue over on the Steam Forums, along with us over here. Second class citizens.

**Comment by mSparks43** [comment]
> Edit: Edited because I did not realize that labels are added at Issue creation. Am creating a separate, more descriptive issue and title that will carry the bug label, see if that gets some movement.
> 
> Edit 2: Rant removed as I have calmed down about it.

Perfectly valid rant (for future reference github stores your edit history, so be careful what you post)
They definitely intentionally broke it, that goes with the territory and how progress happens. What I don't understand is why they keep releasing things that are so broken no one can use them and worse replacing otherwise working versions with ones that don't and are so far from being ready. That slows everything to a crael because no one can get anything done.

Frustrating is putting it mildly.

**Comment by digitalcircuit** [comment]
@mSparks43 I'm quite frustrated by this breakage as well, but considering this SteamVR beta is longer term, and Linux has 0.93% market share that would reasonably run VR (i.e. discounting the Steam Deck: `((100−43.04)÷100)×1.63`), even if we generously assume there are **five times as many** VR users on Linux as there as on Windows, that means that **only 0.07% of Valve's customer base could be impacted by SteamVR beta being broken on Linux.**  And we still have the stable version of SteamVR to revert to during the beta.

```
(((100−43.04)÷100)×1.63)×((1.6÷100)×5)
= 0.07427584
```

> `43.04%` = Linux users running SteamOS "Holo" from https://store.steampowered.com/hwsurvey/?platform=linux
> `1.63%` = Steam users running Linux, including the rather popular Steam Deck (which isn't really suited for VR) from https://store.steampowered.com/hwsurvey/
> `1.6%` = number of Steam users across all platforms with VR from https://store.steampowered.com/hwsurvey/

I'm daily-driving VR on Linux and I want better support from Valve for this, but it feels.. uncharitable to say they *intentionally* broke Linux support in the first month of the beta for a massive overhauling of a niche product (SteamVR) for a niche audience (Linux users on VR).  Even with generous assumptions, we could be a rounding error.

----

For now, back on topic, I'll keep to testing the SteamVR beta with every update released, and when something changes (working or breaks differently) I'll update here with a comment or a thumbs-up if someone else already reported it.

**Comment by mSparks43** [comment]
> @mSparks43 I'm quite frustrated by this breakage as well, but considering this SteamVR beta is longer term, and Linux has 0.93% market share that would reasonably run VR (i.e. discounting the Steam Deck: `((100−43.04)÷100)×1.63`), even if we generously assume there are **five times as many** VR users on Linux as there as on Windows, that means that **only 0.07% of Valve's customer base could be impacted by SteamVR beta being broken on Linux.** And we still have the stable version of SteamVR to revert to during the beta.
> 
> ```
> (((100−43.04)÷100)×1.63)×((1.6÷100)×5)
> = 0.07427584
> ```
> 
> > `43.04%` = Linux users running SteamOS "Holo" from https://store.steampowered.com/hwsurvey/?platform=linux
> > `1.63%` = Steam users running Linux, including the rather popular Steam Deck (which isn't really suited for VR) from https://store.steampowered.com/hwsurvey/
> > `1.6%` = number of Steam users across all platforms with VR from https://store.steampowered.com/hwsurvey/
> 
> I'm daily-driving VR on Linux and I want better support from Valve for this, but it feels.. uncharitable to say they _intentionally_ broke Linux support in the first month of the beta for a massive overhauling of a niche product (SteamVR) for a niche audience (Linux users on VR). Even with generous assumptions, we could be a rounding error.
> 
> For now, back on topic, I'll keep to testing the SteamVR beta with every update released, and when something changes (working or breaks differently) I'll update here with a comment or a thumbs-up if someone else already reported it.

This is all caused by their linux dev platform being so unstable its been virtually impossible to develop for, not a reason for it.

**Comment by TTimo** [comment]
Hello,

The errors from `vrenv.sh` are a red herring and unrelated to the issue here.

- Make sure you do not use snap. On Debian and derivatives, we only support (and recommend) using Steam installed with Valve's official .deb installer: http://repo.steampowered.com/steam/

I don't think it's the case here that anyone is using snap, but it's worth mentioning (again - don't!).

- Make sure you are using the latest beta of the Steam client (also seems to be the case for everyone here).

- Make sure `zenity` is installed.

As I'm writing this I realize there may be a problem and we are not using the zenity bundled in the Steam runtime like we should. This could explain the hang (will fix).

- If you are still having problems after this, please check and upload the content of `/tmp/SteamVRLauncherSetup.log` to help with diagnostic.

**Comment by dylanmtaylor** [comment]
I am not using snap. I will check for zenity and see if that fixes it.

**Comment by dylanmtaylor** [comment]
When attempting to install zenity, I get  

zenity is already the newest version (3.44.2-1).

**Comment by JupiterSky11** [comment]
I also have both Zenity and the non-snap version.
All I can tell is that stuff isn't found by the environment setup script, so I assume that's causing issues.  If not, then it's just steam's binary files not functioning properly.  Perhaps there are some zenity configs that need to be adjusted, but I have no idea what zenity is or what it does.  +1 to it being bundled with Steam though, a self contained system is always more reliable.

**Comment by TTimo** [comment]
Please un-comment the `set -x` line at the top of `vrsetup.sh` and try again:

```
# verbose
set -x
```

The Steam client should produce an output that looks like this:

![steamvr-vrsetup-verbose](https://github.com/ValveSoftware/SteamVR-for-Linux/assets/6637/048f7667-de78-4042-a227-57ec203e1608)

(zenity is a small app for doing prompts and error dialogs)

**Comment by TTimo** [comment]
Better yet, try replacing `vrsetup.sh` with this version:

https://www.dropbox.com/scl/fi/43xug7bo3nn1j1qy71wew/vrsetup.sh?rlkey=7mqscs6a3h94qbrqf5cacl2tk&dl=0

That may fix the zenity problem (assuming that's what it is) and get you to the pkexec privileges prompt:

![steamvr-vrsetup-scout-ldlp](https://github.com/ValveSoftware/SteamVR-for-Linux/assets/6637/2d689991-4b4a-40b6-ba21-4cf68aab2e90)

**Comment by JupiterSky11** [comment]
```
vrstartup.sh[175966]: === Sat Oct 21 21:08:52 EDT 2023 ===
vrstartup.sh[175966]: Steam Linux Runtime: sniper_platform_0.20231016.63422
vrstartup.sh[175966]: exec /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrstartup.sh
vrenv.sh[175966]: exec /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrstartup.sh
vrstartup.sh[175966]: Steam Linux Runtime: sniper_platform_0.20231016.63422
vrstartup.sh[175966]: call /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
vrsetup.sh[175999]: Detected Steam Linux Runtime pressure-vessel launch in sniper_platform_0.20231016.63422
vrsetup.sh[175999]: Relaunching via steam launcher service to host level for vrcompositor setcap configuration.
vrsetup.sh[176011]: exec /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
vrenv.sh[176011]: exec /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
vrstartup.sh[188851]: === Sat Oct 21 22:32:13 EDT 2023 ===
vrstartup.sh[188851]: Steam Linux Runtime: sniper_platform_0.20231016.63422
vrstartup.sh[188851]: exec /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrstartup.sh
vrenv.sh[188851]: exec /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrstartup.sh
vrstartup.sh[188851]: Steam Linux Runtime: sniper_platform_0.20231016.63422
vrstartup.sh[188851]: call /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
vrsetup.sh[188884]: Detected Steam Linux Runtime pressure-vessel launch in sniper_platform_0.20231016.63422
vrsetup.sh[188884]: Relaunching via steam launcher service to host level for vrcompositor setcap configuration.
vrsetup.sh[188896]: exec /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrenv.sh /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
vrenv.sh[188896]: exec /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
vrsetup.sh[188896]: Relaunching under scout LDLP runtime.
vrsetup.sh[188896]: exec /home/jupitersky/.steam/bin/steam-runtime/run.sh /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
vrsetup.sh[188896]: Detected scout LDLP runtime.
vrsetup.sh[188896]: /home/jupitersky/.steam/debian-installation/steamapps/common/SteamVR/bin/linux64/vrcompositor-launcher binary has cap_sys_nice privileges
```

**Comment by TTimo** [comment]
Alright - so not a problem with zenity or the pkexec process, the script reports you already have adequate permissions.

I suspect `steam-runtime-launch-client` does not return when it should. Once you get to the point above where things are stuck, try to issue `killall steam-runtime-launch-client` ?

**Comment by digitalcircuit** [comment]
@TTimo Partial good news - for my system¹, your [update to the `vrsetup.sh` script](https://github.com/ValveSoftware/SteamVR-for-Linux/issues/618#issuecomment-1773973428 ) made SteamVR beta able to launch! :tada:

Unfortunately, I now get the "Failed to connect to window manager" (`497`) error, [which appears to be impacting a number of other folks, too](https://github.com/ValveSoftware/SteamVR-for-Linux/issues/622 ).

I've subscribed to the above bug report, and assume I should wait for steps there (to keep this focused on failure to launch).  However, if you'd like me to do any more testing, let me know.  And thank you for your time!

¹ Kubuntu 22.04 LTS, Steam via `.deb` package

**Comment by ThaSwapMeetPimp** [comment]
For my system also, the update to vrsetup.sh scrip made beta able to launch, and like digitalcircuit, getting 497 error.

**Comment by ClearlyClaire** [comment]
> Alright - so not a problem with zenity or the pkexec process, the script reports you already have adequate permissions.
> 
> I suspect `steam-runtime-launch-client` does not return when it should. Once you get to the point above where things are stuck, try to issue `killall steam-runtime-launch-client` ?

Yes, `steam-runtime-launch-client` doesn't seem to cleanly exit like it should. It seems to just hang, sometimes without significant CPU usage, [sometimes with 100% CPU usage](https://github.com/ValveSoftware/SteamVR-for-Linux/issues/618#issuecomment-1743580550). The edited script returns without hanging, and then SteamVR kind of works, but SteamVR Home doesn't load, the `steam-runtime-launch-client` for steamtours is at 100% and other related `steam-runtime-launch-client` processes get stuck, with SteamVR does not exiting correctly and requiring `steam-runtime-launch-client` processes to be manually killed.

I also get “SteamVR failed to initialize for unknown reasons. (Error: Shared IPC Compositor Init Failed (303) (303))” which may or may not be related, but this does not prevent the dashboard from working.

**Comment by TTimo** [comment]
Hello,

The launch script fixes will ship in a beta update (likely today).

I would like to keep this issue focused on any launch script and early startup problems that may remain. Everything else I see mentioned since my last update is already reported in other issues.

`steam-runtime-launch-client` is expected to wait on the process that was launched before exiting. There is only going to be a problem if said process has exited, and it still doesn't exit. @JupiterSky11 can you provide an update on this? It seemed like that could be the case for you (@ClearlyClaire also, but that process is expected to stay running while SteamVR Home is up for instance).

**Comment by JupiterSky11** [comment]
@TTimo sorry for the delay, college has been rough.

I did a complete reinstall with the updated script and it worked as far as I can tell.  It opens with a 303 error, the headset boots up, the normal interface appears alongside the steamvr settings window, and as per usual there's a black window that is impossible to close or kill.  I will edit this in a bit to report on functions beyond it turning on and not instantly crashing.

**Comment by ClearlyClaire** [comment]
> `steam-runtime-launch-client` is expected to wait on the process that was launched via the Steam service exiting before also exiting. There is only going to be a problem if said process has exited, and `steam-runtime-launch-client` does not.
> 
> @JupiterSky11 can you provide an update on this? It seemed like that could be the case for you (@ClearlyClaire also, but that process is expected to stay running while SteamVR Home is up for instance).

Yeah, this is consistent with `steam-runtime-launch-client` not exiting when it should. The unmodified `vrsetup.sh` script `exec`s into `steam-runtime-launch-client`, the inner script runs to completion, but `steam-runtime-launch-client` does not exit. If I kill it, it progresses to launching SteamVR, which kinda works as described above. But quitting it leaves dangling `steam-runtime-launch-client` processes (and SteamVR is still marked as “running” inside Steam).

If I simplify `vrsetup.sh` down to the following, it exhibits the same issue:
```sh
#!/bin/bash

# verbose
set -x

set -o pipefail
shopt -s failglob
set -u

if [ -z "${PRESSURE_VESSEL_RUNTIME-}" ]; then
    exit 0
else
    if [ -z ${SRT_LAUNCHER_SERVICE_ALONGSIDE_STEAM-} ]; then
        exit 1
    else
        exec steam-runtime-launch-client --alongside-steam "$0" "$@"
        # unreachable
    fi
    exit 0
fi
```

The script should return immediately, but it doesn't:
![image](https://github.com/ValveSoftware/SteamVR-for-Linux/assets/384364/3fbdfd7c-094d-468a-a1f5-c904c450b171)

**Comment by TTimo** [comment]
@ClearlyClaire I see `vrsetup.sh` is still running, as a child of `steam-runtime-launcher-service`. That's the problem - once it exits, `steam-runtime-launch-client` will likely exit cleanly (that's the whole point of the mechanism, it throws the execution at `steam-runtime-launcher-service` and waits).

What's the status of that `vrsetup.sh` process? If it's the modified version you've shown above it doesn't really have any option to get stuck. So maybe it's in a zombie state at that point and waiting to be reaped?

Can you upload a report out of Help -> Steam Runtime Diagnostics in the Steam client? If you don't feel comfortable linking it publicly you can email it to me at ttimo@valvesoftware.com

**Comment by ThaSwapMeetPimp** [comment]
> The launch script fixes will ship in a beta update (likely today).

Still not launching with 2.0.7, if that was the update that was supposed to have the launch script fixes. Replacing vrsetup.sh still works.

**Comment by extraymond** [comment]
Can confirm with vrsetup.sh fix 2.0.7 works, steamvr beta now boots up instead of showing nothing silently.
Still getting the 303 error and with the old ui though.

ubuntu 22.04, kde, x11, steam-beta(deb)

**Comment by digitalcircuit** [comment]
I apologize, I goofed with my prior success report by overlooking file system permissions…

**The updated `vrsetup.sh` from Dropbox didn't actually fix my issue, removing execute permissions did**, i.e. making the script unlaunchable:
```
$ ls -l "$HOME/.local/share/Steam/steamapps/common/SteamVR/bin/"
[…]
-rwxrwxr-x 1 user user    2807 Oct 24 02:46 vrsetup.sh
-rw-rw-r-- 1 user user    2803 Oct 22 00:55 vrsetup.sh.from_dropbox
[…]
```

There's no difference with the new script versus Dropbox script if I `chmod 755` it.  Also, the `Verify integrity of tool files` button does *not* restore execute permissions if the file is otherwise identical (it will if the file content is different).

I've tried @ClearlyClaire 's [modified version of `vrsetup.sh`](https://github.com/ValveSoftware/SteamVR-for-Linux/issues/618#issuecomment-1776019728 ), making sure to grant it execute permissions, and that has the same problem - `vrsetup.sh` becomes a zombie:

```
$  ps aux | grep -e 'vrsetup.sh' | grep -v grep
user        22233  0.0  0.0      0     0 ?        Zs   02:59   0:00 [vrsetup.sh] <defunct>
```

If desired, I can email a copy of my Steam Runtime Diagnostics.

**Comment by ClearlyClaire** [comment]
> @ClearlyClaire I see `vrsetup.sh` is still running, as a child of `steam-runtime-launcher-service`. That's the problem - once it exits, `steam-runtime-launch-client` will likely exit cleanly (that's the whole point of the mechanism, it throws the execution at `steam-runtime-launcher-service` and waits).
> 
> What's the status of that `vrsetup.sh` process? If it's the modified version you've shown above it doesn't really have any option to get stuck. So maybe it's in a zombie state at that point and waiting to be reaped?

It's the modified `vrsetup.sh` with not much in it. It's a zombie process:
```console
claire@swordfish:~$ ps aux | grep vrsetup
claire     23292  0.0  0.0 159200  5888 ?        Sl   13:13   0:00 steam-runtime-launch-client --alongside-steam /home/claire/.steam/debian-installation/steamapps/common/SteamVR/bin/vrsetup.sh
claire     23301  0.0  0.0      0     0 ?        Zs   13:13   0:00 [vrsetup.sh] <defunct>
claire     25502  0.0  0.0   6368  2048 pts/4    S+   13:16   0:00 grep vrsetup
```

> Can you upload a report out of Help -> Steam Runtime Diagnostics in the Steam client? If you don't feel comfortable linking it publicly you can email it to me at [ttimo@valvesoftware.com](mailto:ttimo@valvesoftware.com)

Sent!

**Comment by TTimo** [comment]
Thanks, we identified the problem and are working on a fix.

**Comment by smcv** [comment]
This should be working in the latest Steam client beta, version 1698185488 (2023-10-24). I don't have VR hardware myself, but I was able to reproduce a similar problem on Ubuntu 22.04 without VR, and I've confirmed that it's fixed now.

The key diagnostic for this is that `steam-runtime-launch-client` never exits and there is a zombie process `[vrsetup.sh] <defunct>`, as shown in @ClearlyClaire's most recent comment. This was mainly (only?) a problem on Debian/Ubuntu and their derivatives, where `/bin/sh` is normally `dash`. Systems where `/bin/sh` is `bash` are believed to have been unaffected.

**Comment by LiamDawe** [comment]
Can confirm SteamVR on Kubuntu 23.10 will now actually load up, although it does initially give this error https://github.com/ValveSoftware/SteamVR-for-Linux/issues/623 and now I'm getting the old VR UI so this issue is also back https://github.com/ValveSoftware/SteamVR-for-Linux/issues/614

**Comment by ClearlyClaire** [comment]
I can confirm that after updating and relaunching both the Steam client beta and SteamVR beta, `vrsetup.sh` and other scripts exit as intended, and Steam VR starts.

**Comment by ThaSwapMeetPimp** [comment]
Can confirm that 2.0.8 launches now, will go to other thread for the other continuing issues.

**Comment by Corben78** [comment]
Still not working for me on Ubuntu 20.04 LTS. Same behaviour as mentioned in https://github.com/ValveSoftware/SteamVR-for-Linux/issues/618#issuecomment-1777037928
Tried to link `/bin/sh` to `/bin/bash` which didn't make a difference.

**Comment by TarsiSurdi** [comment]
I'm facing the issue where these files (`vrstartup.sh` & `vrenv.sh`) don't even exist under `/bin`. I have verified the game's files multiple times, switch from beta to stable, to stable from beta, uninstalled it... - everything! But still both of these script's simply aren't there after an install and clicking "Launch" on the client does absolutely nothing :(

**Comment by smcv** [comment]
@TarsiSurdi: If some files are completely missing from `SteamVR/bin` for you, that doesn't match the original report in this issue or any of the recent comments, so I think that would be better reported as a separate issue.

**Comment by TarsiSurdi** [comment]
> @TarsiSurdi: If some files are completely missing from `SteamVR/bin` for you, that doesn't match the original report in this issue or any of the recent comments, so I think that would be better reported as a separate issue.

I'm not creating another issue since I believe this to be related to the Steam Client Beta itself, but I've been able to solve it yesterday. Turns out Steam was downloading the _Windows_ version of SteamVR even when no compatibility tool was selected for the app at all.

After realizing this had been happening I went to its compatibility settings, turned it on by selecting random Proton versions and then turned it off again. After doing that, Steam downloaded the correct _Linux_ specific binaries.

Also, by following advice in this thread I have downgraded to version 1.27 and am now able to launch Steam VR, connect to my headset through ALVR and get into SteamVR home - things I' not able to do with the current "stable" version; launching any applications doesn't seem to work as intended though.

When opening games the dashboard keeps "loading..." then indefinitely, is able to launch them on my computer but no image gets streamed to the HMD. I've tested the following games with regular Proton Experimental and ProtonGE8-17 and all have presented the same behavior:
- Beat Saber;
- Tetris Effect: Connected;
- Keep Talking and Nobody Explodes;
- Half Life: Alyx;

However, I think this might be a problem with my setup since I'm running on GNOME Wayland with NVIDIA 535 drivers. I'll be trying out the latest beta NVIDIA drivers (545) today since it references fixes for VR gaming in its changelog and could possibly be the issue in this situation.

**Comment by vermeeren** [comment]
@TarsiSurdi Could you try with ProtonGE8-16? That version works for me with VRChat in VR. Newer version such as ProtonGE8-21 (and most probably the new Proton experimental) don't work in VR, the game crashes instantly instead. Curious to see if it works for you then.

**Comment by TarsiSurdi** [comment]
> @TarsiSurdi Could you try with ProtonGE8-16? That version works for me with VRChat in VR. Newer version such as ProtonGE8-21 (and most probably the new Proton experimental) don't work in VR, the game crashes instantly instead. Curious to see if it works for you then.

Just tested SteamVR 1.27 and 2.0.8 with the same games as before + Steam 360 Video Player on both GNOME and KDE Plasma using Wayland and X11 (NVIDIA proprietary drivers 545.23.06, Fedora 39 beta). I also set the compatibility tool to ProtonGE8-16, as suggested in this issue.

Version 2.0.8 fails to connect to the window manager no matter the environment used.
Version 1.27 gets me into SteamVR Home. When launching apps / games they'll open up on my PC just fine; I don't get any video in the headset. That's true even with native applications such as the Steam 360 Video Player - I cannot get any video output from any apps EXCEPT SteamVR Home and the void.

**Comment by vermeeren** [comment]
@TarsiSurdi Mmm I feel like I vaguely remember the bug where it gets stuck wanting to update to the Windows version, could explain why it doesn't want to launch games properly too. I believe this happens once you touch force compat tool once, it then permanently bugs.

1. Uncheck any "force compat tool" in Steam for SteamVR.
2. Uninstall SteamVR.
3. With Steam shut down open `~/.steam/debian-installation/userdata/$userid/config/compat.vdf` with a text editor and search for `250820`, if there is an entry remove it completely.
4.Edit `~/.steam/debian-installation/config/config.vdf` again search for `250820`, but only the Proton/compat configuration part in the `CompatToolMapping` array. Remove the entry if it exists.
5. Boot up Steam and install SteamVR again.

Does that fix it for you?

**Comment by kisak-valve** [comment]
Closing as fixed.

@TarsiSurdi, you've described the issue being tracked at https://github.com/ValveSoftware/steam-for-linux/issues/9875.
